torch/csrc/jit/codegen/cuda/interface.cpp

```diff
@@ -48,7 +182,47 @@ RegisterOperators reg({
         },
         c10::AliasAnalysisKind::INTERNAL_SPECIAL_CASE),
 });
-}
+
+RegisterOperators reg_guard({
+    Operator(
+        prim::CudaFusionGuard,
+        [](const Node* node) -> Operation {
+          return [node](Stack* stack) {
+            FUSER_PERF_SCOPE("CudaFusionGuard");
+            // TODO: check latency here!!!!
+            std::vector<TypePtr> types = node->tys(attr::types);
+            const auto num_inputs = types.size();
+            at::ArrayRef<IValue> inputs = last(stack, num_inputs);
+            drop(stack, num_inputs);
+
+            if (!fuser::cuda::getCudaFusionGuardMode()) {
+              push(stack, IValue(true));
+              return;
+            }
+
+            for (size_t i = 0; i < num_inputs; i++) {
+              const c10::TensorTypePtr& guard_tensor_type =
+                  types[i]->cast<TensorType>();
+
+              // TODO: maybe we should just push false and fallback
+              TORCH_INTERNAL_ASSERT(inputs[i].isTensor());
```

Comment:
I think assert is appropriate here until you start checking non tensor properties.

Reply:
Sounds good.
We will start doing that for dynamic reduction axes. I'll just leave it here until next PR.